source/client-side-operations-timeout/tests/README.rst

```diff
@@ -31,10 +31,12 @@ during a test MUST be unset using ``internalClient`` after the test has been exe
 created for tests MUST be configured with read/write concern ``majority``, read preference ``primary``, and command
 monitoring enabled to listen for ``command_started`` events.
 
-Multi-batch writes
-~~~~~~~~~~~~~~~~~~
+1. Multi-batch writes
+~~~~~~~~~~~~~~~~~~~~~
 
-This test MUST only run against server versions 4.4 and higher.
+This test MUST only run against standalones on server versions 4.4 and higher.
+The ``insertMany`` call takes an exceedingly long time on replicasets and sharded
+clusters.
```

Comment:
See the [Go driver PR](https://github.com/mongodb/mongo-go-driver/pull/1081) implementing this prose test. In order for the first `insert` of the `insertMany` call to succeed, the `timeoutMS` value needs to allow for a high operation execution time after the blocking failpoint.

>On a standalone topology, [the first `insert` of the `insertMany` call] takes 274ms; on a replicaset topology, this takes 654 ms; on a sharded topology, this takes 971ms.

These times are even longer on other OSes (I'm on a Mac M2). I'm also suggesting restricting this test to standalones. In order to consistently successfully test against sharded clusters, I had to bump the blocking failpoint to ~3 seconds and `timeoutMS` to 6 seconds. That made for a seriously long prose test. The top of this section does say drivers are allowed to run CSOT prose tests solely against single-node replsets, so maybe this is redundant? Let me know what you think.

Reply:
What if you use w=1 instead of the implicit w:majority default?

Reply:
Using `w=1` seems to shave off about 50-100ms depending on the topology. I could also require that, but I think we should still bump the timeout.

Reply:
> Should we add a note that drivers may need to adjust the timeouts depending on their performance of bulk encoding?

Done.

> Also if we did add the w=1 approach

I'm not sure we need to do suggest `w=1` given that it does not seem to significantly improve performance and drivers are allowed to adjust the timeouts.

Reply:
Another thought. It seems like most of the slowdown is due to having so many documents. Using 50 document 1MB each also results in 2 batches (with OP_MSG payloads 48MB can fit in one batch) but has much lower overhead.

Reply:
Good idea. For the Go driver: 150ms on standalone, 550ms on replset, 1s on sharded replicaset. Modified the prose test. I've still left the requirement of only running against standalones, but let me know if you think I should allow replsets (sharded replicaset may require higher suggested timeouts).

Reply:
Thanks, let's leave it at only standalones.